### PR TITLE
Fix 701

### LINKED ISSUE #701: jwt_client_id versus security.oidc.clientId
I'm not sure if `security.oidc.clientId` and `jwt_client_id` are always the same, or they can be different.

In other words, if we should reuse variable names like @djtfmartin did in this branch: https://github.com/AtlasOfLivingAustralia/ala-install/commit/3a5411bc4ed4f35b27cb03e9ccebebf77734e7ce#diff-a1298dd188946a72c166897d4e28ecbbaab003b82e128cbd14fcb7be09b4024c 
or we should use different variable names like in https://github.com/AtlasOfLivingAustralia/ala-install/pull/699
```
egrep -r "jwt_client_id|jwt_secret" .                                       
./ansible/roles/image-service/templates/config/image-service-config.yml:{% if jwt_client_id is defined %}
./ansible/roles/image-service/templates/config/image-service-config.yml:    client-id: {{ jwt_client_id }}
./ansible/roles/image-service/templates/config/image-service-config.yml:    secret: {{ jwt_secret }}
./ansible/roles/species-list/templates/specieslist-webapp-config.properties:webservice.client-id={{ jwt_client_id }}
./ansible/roles/species-list/templates/specieslist-webapp-config.properties:webservice.client-secret={{ jwt_secret }}
```
Currently #699 is causing this error:
```
TASK [species-list : copy all config.properties] *******************************
Monday 09 October 2023  03:43:53 +0200 (0:00:00.633)       0:01:14.212 ******** 
fatal: [ala-install-test-1]: FAILED! => {"changed": false, "msg": "AnsibleUndefinedVariable: 'jwt_client_id' is undefined"}
```
That suggest that we should adapt all our inventories. But if we reuse variables like in Dave's above patch, this adaptation is not necessary.

### PR DESCRIPTION
**Pull request by vjrj**
This PR fix #701. 

Feel free to propose other thing.

### CODE REVIEW

**Review by djtfmartin — APPROVED** [review]
LGTM

**Comment by joe-lipson** [comment]
There are some large PRs #667 and [ansible-inventories #35](https://github.com/AtlasOfLivingAustralia/ansible-inventories/pull/35) that add these jwt config values well. I've only recently looked at these in any detail, they use different variable names and in some cases different values for oidc and webservice IDs and secrets. There will be some conflicts here

**Comment by vjrj** [comment]
I don't see any conflict. If you want to use different client/secret values, you can still use them.

image

I didn't add this too in `biocache-config.properties` to not cause a merge conflict in the helm branch. 

The named client-id and secret variables (like `lists_client_id`) do not affect ALA because they are only used in the LA community. We use a single inventory for all our services.

**Comment by joe-lipson** [comment]
I mean in [Sushants PR](https://github.com/AtlasOfLivingAustralia/ala-install/pull/667/files#diff-ba3df88974364606c68b0bb3235e3bb6fccb86961ec4c5104b7a39b2dab7a7f1) there are another set of different names for those secrets, I wasn't aware of those at the time I chose the names I used.

specieslist-config

I could add those var names to the list of defaults / fallbacks but I'm mindful that it starts to become a long and messy list. I think I'll change my names to be consistent.

What you mentioned about having the config for all services in one inventory sounds like it would be a good thing

**Comment by vjrj** [comment]
About joining inventories, I recently created a ini parser to detect issues in inventories. It's useful if you want to check some inventory to detect issues, or also to merge some inventories (as we did in the past):

https://github.com/living-atlases/generator-living-atlas/blob/master/lint_ini.py

Sample truncated output:
```
Warning: Option 'biocache_base_url' has a conflicting value. Previous: 'https://registros-es.l-a.site' in section 'all:vars' of la-test-inventory.ini at line 263. Current: 'https://registros-ws-es.l-a.site' in section 'dashboard:vars' of la-test-inventory.ini.
Warning: Option 'collectory_url' has a conflicting value. Previous: 'https://colecciones-es.l-a.site' in section 'all:vars' of la-test-inventory.ini at line 335. Current: 'https://colecciones-es.l-a.site/ws/' in section 'pipelines:vars' of la-test-inventory.ini.
```

Many times some variable have different uses in different roles. This is why I'm adding other role based names (for our single inventory) and a default fallback for ALA ones.

File: ansible/roles/biocache3-properties/templates/biocache-config.properties [hunk 1]
```diff
@@ -464,6 +464,10 @@ security.jwt.enabled={{ jwt_auth_enabled | default('false') }}
 security.jwt.discovery-uri={{ oidc_discovery_url }}
 security.jwt.clientId={{ (biocache_client_id | default(oidc_client_id)) | default('') }}
 
+webservice.client-id = {{ (biocache_client_id | default(oidc_client_id)) | default('') }}
+webservice.client-secret = {{ (biocache_client_secret | default(oidc_client_secret)) | default('') }}
+webservice.jwt-scopes = {{ (biocache_jwt_scopes | default(oidc_jwt_scopes)) | default('openid,email') }}
```
Comment (adam-collins) [inline ansible/roles/biocache3-properties/templates/biocache-config.properties:469]: Remove or refactor. See current `master` branch.

**Review by adam-collins — CHANGES_REQUESTED** [review]

File: ansible/roles/biocache3-properties/templates/biocache-config.properties (hunk 1 above)
Comment (vjrj) [reply ansible/roles/biocache3-properties/templates/biocache-config.properties]: I've just resolved the conflicts with master.

**Review by adam-collins — APPROVED** [review]
approved

**Comment by djtfmartin** [comment]
thanks @vjrj & @adam-collins

**Pull request merged**